### PR TITLE
#8664 Rollback if error. Close session in finally.

### PR DESCRIPTION
**Pull request by dsilvam**

### CODE REVIEW

**Comment by dsilvam** [comment]
3.6 version of https://github.com/dotCMS/core/pull/8665

File: src/com/dotmarketing/business/ajax/RoleAjax.java [hunk 1]
```diff
@@ -692,104 +692,134 @@ public void deleteLayout(String layoutId) throws DotDataException, PortalExcepti
...
-		}
+			HibernateUtil.commitTransaction();
 
-		if(cascade && permissionable.isParentPermissionable()) {
-			Logger.info(this, "Cascading permissions for role " + roleId + " and folder/host id " + folderHostId);
-			Role role = APILocator.getRoleAPI().loadRoleById(roleId);
-			CascadePermissionsJob.triggerJobImmediately(permissionable, role);
-			Logger.info(this, "Done cascading permissions for role " + roleId + " and folder/host id " + folderHostId);
+		} catch (Exception e) {
+			HibernateUtil.rollbackTransaction();
```
Comment (jcastro-dotcms) [inline src/com/dotmarketing/business/ajax/RoleAjax.java]: Maybe we should add some logging here to figure out what caused the error?

Comment (dsilvam) [inline src/com/dotmarketing/business/ajax/RoleAjax.java]: yes @jcastro-dotcms that is what needs to be done

File: src/com/dotmarketing/business/ajax/RoleAjax.java [hunk 2]
```diff
@@ -692,104 +692,135 @@ public void deleteLayout(String layoutId) throws DotDataException, PortalExcepti
...
-		}
+			HibernateUtil.commitTransaction();
 
-		if(cascade && permissionable.isParentPermissionable()) {
-			Logger.info(this, "Cascading permissions for role " + roleId + " and folder/host id " + folderHostId);
-			Role role = APILocator.getRoleAPI().loadRoleById(roleId);
-			CascadePermissionsJob.triggerJobImmediately(permissionable, role);
-			Logger.info(this, "Done cascading permissions for role " + roleId + " and folder/host id " + folderHostId);
+		} catch (Exception e) {
+			Logger.error(this, "Error saving permissions for role " + roleId + " and folder/host id:" + folderHostId, e);
```
Comment (dsilvam) [inline src/com/dotmarketing/business/ajax/RoleAjax.java]: @jcastro-dotcms Thanks for pointing out!

**Comment by jgambarios** [comment]
Closed pull request and created a new one for master-3.7: https://github.com/dotCMS/core/pull/9947

**Pull request closed without merging**